src/Wpf.Ui.Tray/Controls/NotifyIcon.cs

```diff
@@ -466,4 +483,13 @@ private void RegisterHandlers()
         internalNotifyIconManager.MiddleClick += OnMiddleClick;
         internalNotifyIconManager.MiddleDoubleClick += OnMiddleDoubleClick;
     }
+
+    private void OnDataContextChanged(object sender, DependencyPropertyChangedEventArgs e)
+    {
+        // Update ContextMenu DataContext when NotifyIcon DataContext changes
+        if (Menu != null && e.NewValue != null)
```

Comment:
shouldn't the latter part be left out? If `this.DataContext` is now `null`, `Menu.DataContext` should perhaps also become `null`